packages/docusaurus/config/docusaurus/plugins/cli-docs.ts

```diff
@@ -0,0 +1,387 @@
+import type {Options as MDXLoaderOptions, MDXPlugin}          from '@docusaurus/mdx-loader';
+import type {LoadContext, Plugin, PluginContentLoadedActions} from '@docusaurus/types';
+import {YarnVersion, miscUtils}                               from '@yarnpkg/core';
+import type {BaseContext, Definition}                         from 'clipanion';
+import {Cli}                                                  from 'clipanion';
+import glob                                                   from 'fast-glob';
+import jiti                                                   from 'jiti';
+import path                                                   from 'path';
+
+const PLUGIN_NAME = `docusaurus-plugin-yarn-cli-docs`;
+const ROUTE_PREFIX = `/cli`;
+function dedent(value: string) {
+  return value.trim().split(`\n`).map(line => line.trimStart()).join(`\n`);
+}
+
+const loader = jiti(__filename, {
+  cache: true,
+  requireCache: false,
+});
```

Comment:
What is this for? I see you mentioned jiti in the PR's comment, but that you also intended to remove it

Reply:
`jiti` is a library to create loader functions that are drop-in replacements for `require` with on-the-fly TS compilation, used by Docusaurus itself (e.g. to require `docusaurus.config.ts`).

I want the CLI doc plugin to do hot reload. That is, if I change a command's metadata while the dev server is running, I want the change to reflect on the page without restarting the dev server. The problem with that is the `cli-docs.ts` is compiled by Docusaurus's internal `jiti` loader instance, so any `import` or `require` I use will just be intercepted. But that `jiti` instance has an internal, hidden cache, so it will just load the cached stale instance of the changed command file, breaking hot reload. To get around that, I needed to create a separate `jiti` loader instance which properly recompile every time it is called.

The trouble comes when I tried to integrate our own on-the-fly TS compilation pipeline (`scripts/setup-ts-execution`) into that `jiti` loader instance. Turns out `jiti` and `esbuild` have some subtle incompatibilities, so I just removed the `esbuild` integration and use `jiti` with its built-in babel-based compilation pipeline. The intention was never to remove `jiti`.